### PR TITLE
DATAUP-468 cleanup utility modules

### PR DESCRIPTION
**Pull request by briehl**
# Description of PR purpose/changes

Taking a pass at some cleanup around my recent work with cell utility code. I also found a few bits that should be merged and/or removed.

Short version = merged `common/utils` with `common/cellUtils`, as they have a few completely duplicated functions. Like, copied and pasted. There were a few functions in there that also just weren't used at all in this codebase, so they were removed. Remaining functions were migrated, or in the case of `toBoolean` replaced in the few places it was used with the simple `!!x` idiom.

# Jira Ticket / Issue #
https://kbase-jira.atlassian.net/browse/DATAUP-468
- [x] Added the Jira Ticket to the title of the PR (e.g. `DATAUP-69 Adds a PR template`)

# Testing Instructions
* Details for how to test the PR:
* There should be no changes.
- [x] Tests pass locally and in GitHub Actions
- [x] Changes available by spinning up a local narrative and navigating to _X_ to see _Y_

# Dev Checklist:

- [x] My code follows the guidelines at https://sites.google.com/lbl.gov/trussresources/home?authuser=0
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] (JavaScript) I have run Prettier and ESLint on changed code manually or with a git precommit hook

### CODE REVIEW

File: kbase-extension/static/kbase/custom/custom.js [hunk 1]
```diff
@@ -107,7 +107,7 @@ define([
     'base/js/keyboard',
     'notebook/js/keyboardmanager',
     'notebook/js/cell',
-    'common/utils',
+    'common/cellUtils',
```
Comment (briehl) [inline kbase-extension/static/kbase/custom/custom.js:110]: This pattern will pop up over and over - functions that were only in `common/utils` and not `common/cellUtils` were copied over there, so just updating the import (and not what it was loaded as) is safe enough: each function is meant to be executed as a static utility call.

File: kbase-extension/static/kbase/js/common/cellComponents/tabs/infoTab.js [hunk 2]
```diff
@@ -52,7 +50,7 @@ define(['common/format', 'common/utils'], (format, utils) => {
             if (!appSpec && arg.currentApp) {
                 appSpec = model.getItem(`app.specs.${arg.currentApp}`); // for bulk cells
             }
-            const appRef = [appSpec.info.id, model.getItem('app').tag].filter(toBoolean).join('/');
+            const appRef = [appSpec.info.id, model.getItem('app').tag].filter((v) => !!v).join('/');
```
Comment (briehl) [inline kbase-extension/static/kbase/js/common/cellComponents/tabs/infoTab.js:53]: This pattern was present in a few modules:
1. import `utils.toBoolean` as a separate variable.
2. pass that to a `.filter()` call as the filter function.

Most of these modules re-implemented it by copying and pasting the `toBoolean` function 😞 

The more modern one liner `v => !!v` does the same thing. I compared and contrasted with a gamut of inputs and saw no difference, so I just refactored everything else that way.

File: kbase-extension/static/kbase/js/common/lang.js [hunk 3]
```diff
@@ -1,4 +1,6 @@
 define(['bluebird'], (Promise) => {
+    'use strict';
```
Comment (briehl) [inline kbase-extension/static/kbase/js/common/lang.js:2]: No other changes (yet), but I wanted my IDE to be happy.

File: kbase-extension/static/kbase/js/common/unodep.js [hunk 4]
```diff
@@ -1,52 +1,6 @@
 define([], () => {
     'use strict';
 
-    /*
-     * Show elapsed time in a friendly fashion.
-     */
-    function pad(string, width, char, right) {
```
Comment (briehl) [inline kbase-extension/static/kbase/js/common/unodep.js:7]: Not used anywhere.

File: kbase-extension/static/kbase/js/common/unodep.js [hunk 5]
```diff
@@ -1,52 +1,6 @@
...
-        }
-        for (i = 0; i < padLen; i += 1) {
-            padding += char;
-        }
-        if (right) {
-            return string + padding;
-        }
-        return padding + string;
-    }
-    function formatElapsedTime(value, defaultValue) {
```
Comment (briehl) [inline kbase-extension/static/kbase/js/common/unodep.js:28]: Also not used anywhere.

File: kbase-extension/static/kbase/js/util/appCellUtil.js [hunk 6]
```diff
@@ -131,7 +131,7 @@ define(['common/runtime', 'StagingServiceClient'], (Runtime, StagingServiceClien
             return evaluateAppConfig(
                 model.getItem(['app', 'otherParamIds', fileType]),
                 model.getItem(['params', fileType, 'params']),
-                {}, // no particular options for non-file parameters righ tnow
+                {}, // no particular options for non-file parameters right now
```
Comment (briehl) [inline kbase-extension/static/kbase/js/util/appCellUtil.js:134]: Typo fix from PR #2463

File: nbextensions/advancedViewCell/widgets/advancedViewCellWidget.js [hunk 7]
```diff
@@ -143,17 +143,6 @@ define([
             ui.setContent('fatal-error.message', model.getItem('fatalError.message'));
         }
 
-        function showFatalError() {
```
Comment (briehl) [inline nbextensions/advancedViewCell/widgets/advancedViewCellWidget.js:146]: Neither `showFatalError` or `toBoolean` are used anymore.

File: nbextensions/appCell2/widgets/appCellWidget.js [hunk 8]
```diff
@@ -13,7 +13,6 @@ define([
     'kb_service/client/catalog',
     'kb_service/client/narrativeMethodStore',
     'common/pythonInterop',
-    'common/utils',
```
Comment (briehl) [inline nbextensions/appCell2/widgets/appCellWidget.js:16]: This one's slightly more complex as it imports both `common/utils` and `common/cellUtils` and uses their various non-overlapping functions separately. I kept the `cellUtils` part and adjusted the usage throughout the cell.

File: kbase-extension/static/kbase/js/common/unodep.js [hunk 9]
```diff
@@ -108,8 +62,6 @@ define([], () => {
     }
 
     return {
-        formatElapsedTime: formatElapsedTime,
-        formatTime: formatTime,
-        isEqual: isEqual,
+        isEqual,
```
Comment (briehl) [inline kbase-extension/static/kbase/js/common/unodep.js:65]: All that's left now in this module is `isEqual`, which should be combined with the `lang` module, as I think that's intended to adjust some language features?

File: nbextensions/editorCell/widgets/editors/readsSet/editor.js [hunk 10]
```diff
@@ -156,10 +156,6 @@ define([
             ui.setContent('fatal-error.advice', editorState.getItem('fatalError.advice'));
         }
 
-        function showFatalError(arg) {
```
Comment (briehl) [inline nbextensions/editorCell/widgets/editors/readsSet/editor.js:159]: Never used in this module.

File: kbase-extension/static/kbase/js/common/cellComponents/tabs/infoTab.js (hunk 2 above)
Comment (ialarmedalien) [reply kbase-extension/static/kbase/js/common/cellComponents/tabs/infoTab.js:53]: 😖

**Comment by briehl** [comment]
The SonarCloud "bug" is from setting `cell.metadata` to itself in one of the modules - this is on purpose to trigger an event in the setter.

2 of the code smells are from using `require` (too many inputs to an arrow function, but these are dependencies!), and the other is in `unodep` which will get fixed in the next (smaller....) PR.

**Review by ialarmedalien — APPROVED** [review]
LGTM!

**Pull request merged**